src/build.rs

```diff
@@ -259,17 +259,58 @@ impl BuildQueue {
             cmd.env("CARGO_TARGET_DIR", &Path::new("target").join("rls"));
             cmd.current_dir(build_dir);
 
-            let mut new_cmd_line = match cmd.output() {
+            let new_cmd_line_args = match cmd.output() {
                 Ok(output) => {
                     let out = String::from_utf8(output.stderr).unwrap();
                     // println!("output: `{}`", out);
                     let exit_str = "     Running `";
+
                     match out.rfind(exit_str) {
                         Some(i) => {
                             let remaining = &out[i + exit_str.len() ..];
                             let end = remaining.find('`').unwrap();
-                            remaining[..end].to_owned()
+                            let remaining = &remaining[..end];
+                            let split = remaining.split(' ');
+
+                            let mut result = vec![];
+                            let mut in_quoted_arg = false;
+
+                            for element in split {
```

Comment:
That seems not too bad. I'd allocate something like 64 bytes or more (we don't expect very many strings and they are long-lived, so memory over-use is not a problem). I imagine very large args are rare. This code is not very hot either, so I think the simplest solution is probably the best.